express/dataset.py

```diff
@@ -0,0 +1,209 @@
+            subset_columns = [f"{name}_{field}" for field in fields]
+            for col in subset_columns:
+                if col not in output_df.columns:
+                    raise ValueError(
+                        f"Column {col} present in output subsets but not found in dataset"
+                    )
+
+            # load subset dataframe
+            subset_df = output_df[subset_columns]
+            # add to the manifest and upload
+            self._upload_subset(name, subset.fields, subset_df)
+
+    def upload(self, save_path):
+        Path(save_path).parent.mkdir(parents=True, exist_ok=True)
+        self.manifest.to_file(save_path)
+        return None
+
+
+class FondantComponent:
+    type: str = "transform"
+
+    @classmethod
+    def _load_spec(cls) -> ExpressComponent:
+        # note: Fondant spec always needs to be called like this
+        # and placed in the src directory
+        spec_path = "fondant_component.yaml"
+        return ExpressComponent(spec_path)
+
+    @classmethod
+    def run(cls) -> dd.DataFrame:
+        """
+        Parses input data, executes the transform, and creates output artifacts.
+
+        Returns:
+            Manifest: the output manifest
+        """
+        # step 1: load component spec
+        spec = cls._load_spec()
+        # step 2: add and parse arguments
+        args = cls._add_and_parse_args(spec)
+        # step 3: create Fondant dataset based on input manifest
+        metadata = json.loads(args.metadata)
+        if cls.type == "load":
+            # TODO ideally get rid of arrs.metadata
+            # by including them in the storage args,
+            # getting run_id based on args.output_manifest_path
+            manifest = Manifest.create(
+                project_name=metadata["project_name"],
+                base_path=metadata["base_path"],
+                run_id=metadata["run_id"],
+                component_id=metadata["component_id"],
+            )
+        else:
+            manifest = Manifest.from_file(args.input_manifest_path)
+        dataset = FondantDataset(manifest)
+        # step 4: transform data
+        if cls.type == "load":
+            output_df = cls.load(args)
+            dataset.add_index(output_df)
+            dataset.add_subsets(output_df, spec)
+        else:
+            # create HF dataset, based on component spec
+            input_dataset = dataset.load_data(spec)
+            # provide this dataset to the user
+            output_df = cls.transform(
+                dataset=input_dataset,
+                args=args,
+            )
+
+        # step 5: create output manifest
+        output_manifest = dataset.upload(save_path=args.output_manifest_path)
+
+        return output_manifest
+
+    @classmethod
+    def _add_and_parse_args(cls, spec):
+        """
+        Add and parse component arguments based on the component specification.
+        """
+        parser = argparse.ArgumentParser()
```

Comment:
Do we need to define these arguments since we're building on top of KfP 1.X? Shouldn't we build on top of 2.X?

Reply:
Does KubeFlow 2.X not require arguments?

Reply:
Let's upgrade in a separate PR.

Reply:
Ok!